lib/jxl/dec_frame.cc

```diff
@@ -737,6 +740,31 @@ Status FrameDecoder::ProcessSections(const SectionInfo* sections, size_t num,
   }
   if (has_error) return JXL_FAILURE("Error in DC group");
 
+  bool do_progressive = false;
+  if (pause_at_progressive_) {
```

Comment:
Shouldn't we have a check about kRegularFrame here too?

Reply:
sure. I added the check later, but it makes sense to check as early as possible.